packages/eventual-send/src/index.d.ts

```diff
@@ -130,7 +144,13 @@ interface EProxy {
    * @param {*} x target for method/function call
    * @returns {ESingleCall} method/function call proxy
    */
-  <T>(x: T): ESingleCall<Unpromise<T>>;
+  <T>(x: T): ESingleCall<
+    T extends Remotable<infer U>
```

Comment:
This is the first mention of `U`. What is it inferred from?

Needless to say, since I don't understand the things this is based on, several levels deep, I am not able to decypher this. At least in this case I feel fairly confident of what `EProxy` is supposed to mean.

Reply:
As above, this type should use `RemoteInterfaceOf` instead of `Remotable`:

> This is the first mention of `U`. What is it inferred from?

It is inferred as the type `U` such that `T extends RemoteInterfaceOf<U>`.  So, `T is the remote interface of some U`.

```js
T extends RemoteInterfaceOf<infer U> // if a given T is the remote interface of some U
   ? // then,
  FunctionOnly<U> // use all the function properties of that U
  : // otherwise,
  Awaited<T> extends RemoteInterfaceOf<infer U> // if the final resolution of T is the remote interface of some U
  ? // then,
  FunctionOnly<U> // use the function properties of that U
  : // otherwise,
  Awaited<T> // use the final resolution of that T
```